xonsh/dirstack.py

```diff
@@ -64,6 +66,22 @@ def cd(args, stdin=None):
                     return '', e.format(len(DIRSTACK))
                 else:
                     d = DIRSTACK[num - 1]
+            else:
+                # NOTE: this CDPATH implementation differs from the bash one.
+                # In bash if a CDPATH is set, an unqualified local folder
+                # is considered after all CDPATHs, example:
+                # CDPATH=$HOME/src (with src/xonsh/ inside)
+                # $ cd xonsh -> src/xonsh (whith xonsh/xonsh)
+                # a second $ cd xonsh has no effects, to move in the nested xonsh
+                # in bash a full $ cd ./xonsh is needed.
+                # In xonsh a relative folder is allways preferred.
+                def tryCDPATH(d):
```

Comment:
PEP8 names for functions should be all lower case with underscores.

Also I don't think that this function needs to be defined inside of the cd() function.  You should probably move this to global scope and pass in cdpaths.